posthog/tasks/process_event.py

```diff
@@ -194,6 +195,11 @@ def process_event(
         if data.get("$set"):
             _update_person_properties(team_id=team_id, distinct_id=distinct_id, properties=data["$set"])
 
+    if not Event.objects.filter(team_id=team_id).exists():
```

Comment:
Maybe it'd be better to add a boolean field on Team for whether an event has been captured yet? False by default, and just switch to True when processing any event. Almost zero performance impact, since we're already selecting Team in `_capture` (in this file even). Would also avoid having to query `team.event_set.exists()` in `posthog/api/user.py`.